aesara/tensor/extra_ops.py

```diff
@@ -1625,6 +1625,29 @@ def infer_shape(self, fgraph, node, ins_shapes):
 broadcast_to_ = BroadcastTo()
 
 
+def geomspace(start, end, steps, base=10.0):
+    from aesara.tensor.math import log
+
+    start = at.as_tensor_variable(start)
+    end = at.as_tensor_variable(end)
+    return base ** linspace(log(start) / log(base), log(end) / log(base), steps)
+
+
+def logspace(start, end, steps, base=10.0):
+    start = at.as_tensor_variable(start)
+    end = at.as_tensor_variable(end)
+    return base ** linspace(start, end, steps)
+
+
+def linspace(start, end, steps):
+    start = at.as_tensor_variable(start)
+    end = at.as_tensor_variable(end)
+    arr = at.arange(steps)
+    arr = arr.dimshuffle([0] + ["x"] * max(start.ndim, end.ndim))
```

Comment:
This one is a bit hard to read. Is it equivalent to  the following?


```suggestion
    arr = at.shape_padright(max(start.ndim, end.ndim))
```

Reply:
Maybe it was padleft?